scripts/build_test.py

```diff
@@ -28,22 +29,144 @@
 
 # pylint: enable=relative-import
 
+
+TEST_DIRECTORY = os.path.join('core', 'tests', 'build', '')
+
+ASSETS_DEV_DIR = os.path.join(TEST_DIRECTORY, 'assets', '')
+ASSETS_OUT_DIR = os.path.join(TEST_DIRECTORY, 'static', 'assets', '')
+
+EXTENSIONS_DEV_DIR = os.path.join(TEST_DIRECTORY, 'extensions', '')
+
+TEMPLATES_DEV_DIR = os.path.join(TEST_DIRECTORY, 'templates', '')
+BASE_JS_TARGET_DIRPATH = os.path.join(TEMPLATES_DEV_DIR, 'pages', '')
+
+# The source directory should contain files with these extensions.
+ASSETS_FILE_EXTENSIONS = ('.svg', '.json', '.html', '.js',)
+EXTENSIONS_FILE_EXTENSIONS = ('.py', '.html', '.js',)
+
+INVALID_INPUT_FILEPATH = os.path.join(
+    TEST_DIRECTORY, 'invalid', 'path', 'to', 'input.js')
+INVALID_OUTPUT_FILEPATH = os.path.join(
+    TEST_DIRECTORY, 'invalid', 'path', 'to', 'output.js')
+
+EMPTY_DIRECTORY = os.path.join(TEST_DIRECTORY, 'empty', '')
+
+
+def ensure_directory_exists(file_path):
```

Comment:
Ah, no, too much logic in test files still. Test actions should be deterministic against a fixed background.

Could you set up the core/tests/build directory so that it has all the right files in the right places to start with? Then maybe in the test class intro you can describe the contents of the test dir. But after that you should just run the main functions under test directly. At most, setUp() should just assert that the core/tests/build dir contains what you want it to contain.

If you want to keep things clean, have a dir that's untouched by the test and that's kept under source control (maybe core/tests/build_sources or a subdir within core/tests/build) so that any changes show up in git. And have a separate dir (maybe core/tests/build or a subdir within that, respectively to the choices in the previous parens) that is basically the "play env" for tests. Remember to scrub that dir on tearDown() so that everything is as it was before.